lib/sensors/accelerometer.hpp

```diff
@@ -50,21 +51,37 @@ constexpr std::uint8_t kExpectedDeviceIdValue = 0x44;
 
 class Accelerometer : public II2cMuxSensor<core::RawAccelerationData> {
  public:
-  Accelerometer(core::ILogger &logger, io::II2c &i2c, const std::uint8_t channel);
+  static std::optional<Accelerometer> create(core::ILogger &logger,
+                                             std::shared_ptr<io::II2c> i2c,
+                                             const std::uint8_t channel,
+                                             const std::uint8_t device_address
+                                             = kDefaultAccelerometerAddress);
   ~Accelerometer();
 
-  core::Result configure();
+  /*
```

Comment:
Is this comment necessary? I feel like the function name makes it's purpose quite obvious?

Reply:
did it for completeness sake since I wanted to mostly highlight what the return values mean

Reply:
But if any of the return types or implementation changes, is anyone really going to think to update this comment given that it seems quite self explanatory and you can see the return type really quickly just by looking at the function definition...

Reply:
i see that for the @brief part but the @return surely needs to be clarified

Reply:
Fair enough